landside/catkin_ws/src/gui/src/gui/launch_dialog.py

```diff
@@ -102,49 +117,200 @@ def node_name_selected(self, item_index):
         selected_node = self.node_name_box.currentText()
         selected_node_file_type = selected_node.split(".")[1]
 
-        if selected_node_file_type == "launch":
-            package_dir = os.path.join(self.ROOT_PATH, self.package_name_box.currentText())
-            launch_file_path = os.path.join(package_dir, 'launch/' + self.node_name_box.currentText())
+        if selected_node_file_type != "launch":
+            return
+
+        # Selected node file is a launch file
+        self.get_args_from_launch_file()
+
+        for row, arg in enumerate(self.arg_form_rows):
+            arg['allow_empty'] = True
+
+            default_value = '' if arg.get('default') is None else arg['default']
+
+            label = QtWidgets.QLabel(arg['name'])
+
+            input = QtWidgets.QLineEdit()
+            input.setText(default_value)
+
+            toolTip = ""
+
+            if arg.get("doc") is not None:
+                doc_dict = {}
+                try:
+                    doc_dict = json.loads(arg["doc"])
+                except Exception:
+                    rospy.logwarn(f"Could not parse JSON doc string for argument `{arg['name']}` in "
+                                  f"`{selected_node}`. Defaulting to unrestricted string input.")
+
+                if doc_dict.get("options") is not None:
+                    input = self.interpret_options(selected_node, arg, doc_dict, input)
+
+                elif doc_dict.get("regex") is not None:
+                    input, toolTip = self.interpret_regex(selected_node, arg, doc_dict, input, toolTip)
+
+                elif doc_dict.get("type") is not None:
+                    input, toolTip = self.interpret_type(selected_node, arg, doc_dict, input, toolTip)
+
+                if doc_dict.get("help") is not None:
+                    if isinstance(doc_dict["help"], str) and doc_dict["help"].strip():
+                        if toolTip:
+                            toolTip += " | Help: "
+                        toolTip += doc_dict["help"]
+
+                if doc_dict.get("allowEmpty") is not None:
+                    if type(doc_dict["allowEmpty"]) == bool:
```

Comment:
Should you also be using `isinstance` here, since you did above for strings?

Reply:
Replaced `type` with `isinstance`.